src/google/protobuf/repeated_field_unittest.cc

```diff
@@ -1788,18 +1814,46 @@ TEST_F(RepeatedFieldIteratorTest, MutableIteration) {
   EXPECT_EQ(2, *(proto_array_.end() - 1));
 }
 
+TEST_F(RepeatedFieldIteratorTest, MutableReverseIteration) {
+  RepeatedField<int>::reverse_iterator iter = proto_array_.rbegin();
+  EXPECT_EQ(2, *iter);
+  ++iter;
+  EXPECT_EQ(1, *iter++);
+  EXPECT_EQ(0, *iter);
+  ++iter;
+  EXPECT_TRUE(proto_array_.rend() == iter);
+
+  EXPECT_EQ(0, *(proto_array_.rend() - 1));
+}
+
 TEST_F(RepeatedFieldIteratorTest, ConstIteration) {
   const RepeatedField<int>& const_proto_array = proto_array_;
   RepeatedField<int>::const_iterator iter = const_proto_array.begin();
+  EXPECT_TRUE(const_proto_array.cbegin() == iter);
   EXPECT_EQ(0, *iter);
   ++iter;
   EXPECT_EQ(1, *iter++);
   EXPECT_EQ(2, *iter);
   ++iter;
   EXPECT_TRUE(const_proto_array.end() == iter);
+  EXPECT_TRUE(const_proto_array.cend() == iter);
   EXPECT_EQ(2, *(const_proto_array.end() - 1));
 }
 
+TEST_F(RepeatedFieldIteratorTest, ConstReverseIteration) {
+  const RepeatedField<int>& const_proto_array = proto_array_;
+  RepeatedField<int>::const_reverse_iterator iter = const_proto_array.rbegin();
+  EXPECT_TRUE(const_proto_array.crbegin() == iter);
```

Comment:
We should also verify that crbegin on a non-const instance returns a const_reverse_iterator.